src/graphql/trusted_domain.rs

```diff
@@ -63,6 +63,29 @@ impl TrustedDomainMutation {
         Ok(name)
     }
 
+    /// Update a trusted domain, returning the new value.
+    #[graphql(guard = "RoleGuard::new(Role::SystemAdministrator)
+        .or(RoleGuard::new(Role::SecurityAdministrator))")]
+    async fn update_trusted_domain(
+        &self,
+        ctx: &Context<'_>,
+        old: TrustedDomainInput,
+        new: TrustedDomainInput,
+    ) -> Result<String> {
+        let name = {
+            let store = crate::graphql::get_store(ctx).await?;
+            let map = store.trusted_domain_map();
+            let old = review_database::TrustedDomain::from(old);
+            let new = review_database::TrustedDomain::from(new);
+            map.update(&old, &new)?;
+            new.name
+        };
```

Comment:
It seems that your style, where a variable is declared early and assigned a specific value later, is still being used. I believe this approach is not ideal. Moreover, in this context, the `name` variable doesn't appear to be essential.

Reply:
Do you have any specific expectations for the return type of the GraphQL API?  
It seems that `Result<()>` cannot be used.  
```
42 | #[Object]
   | ^^^^^^^^^ the trait `OutputType` is not implemented for `()`
```
Are you considering declaring and using a custom scalar type like `Void`?

Reply:
I apologize for my confusion. I thought this was similar to what you had done before, but it seems inevitable for the later asynchronous code. Once again, I’m sorry for my mistake in not catching the later part.